examples/src/main/java/com/mapbox/navigation/examples/core/FreeDriveNavigationActivity.kt

```diff
@@ -0,0 +1,237 @@
+                this,
+                Utils.getMapboxAccessToken(this)
+        )
+
+        mapboxNavigation = MapboxNavigation(
+                applicationContext,
+                Utils.getMapboxAccessToken(this),
+                mapboxNavigationOptions,
+                locationEngine = getLocationEngine()
+        ).also {
+            it.registerRouteProgressObserver(routeProgressObserver)
+        }
+
+        initListeners()
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap)
+            mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+            navigationMapboxMap?.initializeLocationComponent()
+            mapInstanceState?.let { state ->
+                navigationMapboxMap?.restoreFrom(state)
+            }
+            initLocationEngine()
+        }
+    }
+
+    fun initLocationEngine() {
+        val requestLocationUpdateRequest =
+                LocationEngineRequest.Builder(DEFAULT_INTERVAL_IN_MILLISECONDS)
+                        .setPriority(LocationEngineRequest.PRIORITY_NO_POWER)
+                        .setMaxWaitTime(DEFAULT_MAX_WAIT_TIME)
+                        .build()
+
+        mapboxNavigation?.locationEngine?.requestLocationUpdates(
+                requestLocationUpdateRequest,
+                locationListenerCallback,
+                mainLooper
+        )
+        mapboxNavigation?.locationEngine?.getLastLocation(locationListenerCallback)
+    }
+
+    private val routesReqCallback = object : RoutesRequestCallback {
+        override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            if (routes.isNotEmpty()) {
+                navigationMapboxMap?.drawRoute(routes[0])
+                if (shouldSimulateRoute()) {
+                    (mapboxNavigation?.locationEngine as ReplayRouteLocationEngine).assign(routes[0])
+                }
+                startNavigation.visibility = View.VISIBLE
+            } else {
+                startNavigation.visibility = View.GONE
+            }
+        }
+
+        override fun onRoutesRequestFailure(throwable: Throwable, routeOptions: RouteOptions) {
+            Timber.e("route request failure %s", throwable.toString())
+        }
+
+        override fun onRoutesRequestCanceled(routeOptions: RouteOptions) {
+            Timber.d("route request canceled")
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    fun initListeners() {
+        startNavigation.setOnClickListener {
+            navigationMapboxMap?.enterNavigationMode()
+            navigationMapboxMap?.addProgressChangeListener(mapboxNavigation!!)
+            if (mapboxNavigation?.getRoutes()?.isNotEmpty() == true) {
+                navigationMapboxMap?.startCamera(mapboxNavigation?.getRoutes()!![0])
+            }
+            mapboxNavigation?.startTripSession()
```

Comment:
When we start the trip session, we should stop the local location updates, otherwise, they're going to compete.

Reply:
thanks, I didn't know this.